fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergCatalog.java

```diff
@@ -238,20 +239,46 @@ default List<Pair<List<Object>, Long>> getSamples() {
         return new ArrayList<>();
     }
 
+    default Map<String, Partition> getPartitionsByDataFiles(String dbName, String tableName, ExecutorService executorService,  Map<String, Partition> partitionMap) {
```

Comment:
you can just mark it as private.  I don't think it should be overrided by subclasses.

Reply:
> you can just mark it as private. I don't think it should be overrided by subclasses.

Got it. Thanks